### ISSUE TITLE
Multimedia Player - Video: How to distinguish controls for multiple videos on a page

### ISSUE THREAD

**Issue by dfarough** [wet-boew/wet-boew#4775]
@wet-boew/access-working-group 
How can we Indicate which video is being controlled by a given set of controls?  In the past I have by accident started more than one video and then had to guess at which Play /pause button would stop the video. Would use of a field set for the video be a good  way to do this?
Or
. I am thinking of something like Preceding each button in the set with a number representing the position of the video on the page. Probably separating the number and the button name with a Colin would be a good Idea so that the synthesizer could pause between the number and the name.

**Comment by hsrudnicki** [comment]
Are you ready to close the issue or otherwise ?

**Comment by dfarough** [comment]
@wet-boew/access-working-group 
There is still no way to know which video is being controlled by the player controls when there is more than one multi media player on the page.

When using Jaws and Internet explorer 10 adjusting the volume slider changes the progress of the video not the volume.  This does not occur when using Firefox.

This issue should not be closed in my opinion.

**Comment by LaurentGoderre** [comment]
@dfarough for controlling the volume, you are using the up/down arrows right?

**Comment by dfarough** [comment]
yes I am using up and down arrows.  this does seem to work with Firefox though.

**Comment by LaurentGoderre** [comment]
very strange

**Comment by hsrudnicki** [comment]
@dfarough and others : Please update this issue's status whether this issue had been fixed and ready to be closed or otherwise, still needs be open? Thank you

**Comment by hsrudnicki** [comment]
Close this issue, just labeled "Going Stale" or otherwise? Please update this issue's status, thank you

**Comment by hsrudnicki** [comment]
Due to no further discussion/status updates, this issue is closed.

**Comment by mysticjeff** [comment]
Should it beÉ

-----Original Message-----

**Comment by hsrudnicki** [comment]
@LaurentGoderre : Again, please help in making the decision whether this issue keep closed or re-opened?

**Comment by SBleeks** [comment]
If the ARIA label is associated with the UI controls (play/pause for example), the association isn't being recognized by JAWS or NVDA.  For pages with multiple videos, the could the ARIA describedby or labeledby be used on the UI controls to correspond with the heading of the video.

As far as volume control, I am able to increase and decrease volume with the up and down arrows using IE11 JAWS 16 and NVDA latest build.

**Comment by dfarough** [comment]
I think that using aria-labeled or aria described by would be a good option for these controls.

The issue here is primarily related to context.  It would be quite easy and perhaps desirable to have multiple instances  of the media player on a page.  I think that it might be an idea to decide how this should be handled.  Perhaps some option can be designed  to allow for multiple videos on a page.

If we found that Aria described by or Labeled by was a viable option for associating these controls, we could then decide whether all controls within the player, or perhaps only the first control of the player should use this association so as to reduce the verbosity for screen reader users.  Whatever we do, we should be consistent so that users will know what to expect with our player

The other thing we should consider is the possibility that this content might well exist on a page containing a carousel,  so we need to be sure that the user knows what the play button controls.

**Comment by mysticjeff** [comment]
Yes, sounds good.

Maybe have the end result look something like "Play video: " + [TITLE] as the label for the button 

Like that the extra verbosity doesn't hamper access to the most important part of the message.




-----Original Message-----

**Comment by LaurentGoderre** [comment]
But what if you access the control going backwards on the page?

**Comment by SBleeks** [comment]
That is why the UI control should be associated with the heading/title of the video. This way it won’t matter if your accessing the player from above or below.  The ARIA association will offer the necessary context for the player control.

**Comment by dfarough** [comment]
As long as we are consistent it does not matter in my opinion.  Screen reader users are able to bring up a list of elements which should expose this information. so we could decide to do this for the play button only if we wanted.

**Comment by mysticjeff** [comment]
As long as the controls contain it in the order present 
i.e. play video: + title

the verbiage wouldn't be too bad? 


-----Original Message-----

**Comment by hsrudnicki** [comment]
Based on the recent above discussion, you have reach an agreement in resolving of this issue but can be this issue closed or otherwise?

**Comment by dfarough** [comment]
If these issues have not been addressed, then I see no great need to close this or any other issue raised here.  This also applies to the datepicker  and other wet components.

**Comment by hsrudnicki** [comment]
Remain this issue still addressed/opened or otherwise? Either way, please update again this issue's status, whether this issue can be closed as resolved or otherwise?  Thank you

**Comment by hsrudnicki** [comment]
Can be this issue closed as resolved/fixed or if otherwise? Please update this issue's status, thanks

**Comment by SBleeks** [comment]
@LaurentGoderre 
@LaurentGoderre @masterbee 
Can ARIA labelledby or describedby be implemented as outlined in the comments above?

**Comment by hsrudnicki** [comment]
Is this still an open issue/question? Please update the status of this again, whether this can be closed or otherwise? Thank you

**Comment by SBleeks** [comment]
Yes.


Stacy Bleeks

Web Accessibility Expert - Centre of Expertise on Accessibility | Communications and Dissemination Branch
Statistics Canada | Government of Canada
Stacy.Bleeks@canada.ca<mailto:Stacy.Bleeks@canada.ca> / Tel: 613-951-6137

Expert en accessibilité  - Centre d'expertise en accessibilité | Direction des communications et de la diffusion
Statistique Canada | Gouvernement du Canada
Stacy.Bleeks@canada.ca<mailto:Stacy.Bleeks@canada.ca> / Tél : 613-951-6137

**Comment by masterbee** [comment]
closed since addressed in https://github.com/wet-boew/wet-boew/pull/7507
